jupyter/notebooks/Data Traversal.ipynb

```diff
@@ -0,0 +1,583 @@
+    "            \"_id\":\"desc\", #descending order\n",
+    "        },\n",
+    "    }\n",
+    ")\n",
+    "\n",
+    "first_result = response['hits']['hits'][0]\n",
+    "print(first_result['_id'])\n",
+    "#checking the image\n",
+    "iiif_url = first_result['_source']['state']['derivedData']['thumbnail']['url']\n",
+    "image_url = str(IIIFImageClient().init_from_url(iiif_url).size(width=500))\n",
+    "Image.open(BytesIO(httpx.get(image_url).content))\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e93d1400-0e77-46a1-87cb-2af53a104a2e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\n",
+    "    f\"Found {response['hits']['total']['value']} \"\n",
+    "    f\"results in {response['took'] / 1000}s\"\n",
+    ")\n",
+    "\n",
+    "print()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7efb25b2-f61f-44bc-b7ef-74c10b9d4ff8",
+   "metadata": {},
+   "source": [
+    "This is weird. Why its still getting 10k results? There should only be 20."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1b996359-c8b3-49c2-aac4-1d13e63d337f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(len(response['hits']['hits']))\n",
+    "last_result = response['hits']['hits'][20]\n",
+    "print(last_result['_id'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6c4c9bc7-3ac6-4f70-95cc-f410d7eebd78",
+   "metadata": {},
+   "source": [
+    "The hits does only have 20 elements. I'm confused. Is there any tool that I can use check the response JSON in a human eye frinedly way? (not the leeeeengthy data XD)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c6e78411-5f23-428c-aa4b-03b5c18d7712",
+   "metadata": {},
+   "source": [
+    "Any way, its functionally working. Time to load the process in a function!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d3b56bf4-e7e2-4e95-addc-9f097f125b79",
+   "metadata": {},
+   "source": [
+    "## Functions and Script"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "53f930fc-2d6a-4dae-a500-2de28e4e05fa",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "from elasticsearch import Elasticsearch\n",
+    "from piffle.iiif import IIIFImageClient \n",
+    "from io import BytesIO\n",
+    "import httpx\n",
+    "from PIL import Image\n",
+    "\n",
+    "#connect to ElasticSearch\n",
+    "local_es = Elasticsearch(\n",
+    "    hosts=os.environ['LOCAL_HOST'],\n",
+    "    http_auth=(\n",
+    "        os.environ['LOCAL_USER'],\n",
+    "        os.environ['LOCAL_PASS']\n",
+    "    )\n",
+    ")\n",
+    "\n",
+    "#create a new index\n",
+    "local_es.indices.create(\n",
+    "    index='id_url_index', \n",
+    "    body={\n",
+    "        \"settings\" : {\n",
+    "            \"refresh_interval\": -1,\n",
+    "            \"number_of_replicas\": 0\n",
+    "        },\n",
+    "    }\n",
```

Comment:
You're creating an index which has its settings optimised for bulk reindexing here! 

Because data will be added in at a manageable pace, you should be able to remove this `body` section and use the default settings - that way you won't need to change them later on and you'll be able to use the index for search immediately